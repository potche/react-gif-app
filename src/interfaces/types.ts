export interface INewCategory {
  onNewCategory: (newCategory: string) => void;
}

export interface ICategory {
  category: string;
}

export interface IGifData {
  id: string;
  title: string;
  url: string;
}
