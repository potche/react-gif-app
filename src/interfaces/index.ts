export * from './gifs';
export * from './types';
