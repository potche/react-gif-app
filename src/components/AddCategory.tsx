import { useState } from 'react';
import { type INewCategory } from '../interfaces';

export const AddCategory: React.FC<INewCategory> = ({ onNewCategory }) => {
  const [inputValue, setInputValue] = useState('');

  const onInputChange = (event: React.ChangeEvent<HTMLInputElement>) => {
    setInputValue(event.target.value);
  };

  const onSubmit = (event: React.FormEvent<HTMLFormElement>) => {
    event.preventDefault();
    console.log({ inputValue });

    const category = inputValue.trim().toLowerCase();
    if (category.length > 1) {
      onNewCategory(category);
      setInputValue('');
    }
  };

  return (
    <form onSubmit={(event) => onSubmit(event)}>
      <input
        type="text"
        placeholder="Buscar Gifs"
        value={inputValue}
        onChange={onInputChange}
      />
    </form>
  );
};

export default AddCategory;
