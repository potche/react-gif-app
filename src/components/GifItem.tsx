import { IGifData } from '../interfaces';

export function GifItem({ url, title }: IGifData) {
  return (
    <div className="card">
      <img src={url} alt={title} />
      <p>{title}</p>
    </div>
  );
}
