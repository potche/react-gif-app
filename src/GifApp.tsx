import { useState } from 'react';
import AddCategory from './components/AddCategory';

export function GifApp(): JSX.Element {
  const [categories, setCategories] = useState(['One Punch', 'Deamon Slayer']);

  const onAddCategory = (event: React.MouseEvent) => {
    console.log({event})
    setCategories(['Dragon Ball', ...categories ])
  }

  return (
    <>
      <h1>Hola mundo..</h1>
      <h2>Cruel</h2>

      <AddCategory />
      <ol>
        {categories.map((c) => (
          <li key={ c }>{ c }</li>
        ))}
      </ol>
    </>
  );
}

export default GifApp;
