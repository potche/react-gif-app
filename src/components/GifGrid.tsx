import { GifItem } from './GifItem';
import { useFetchGifs } from '../hooks/useFetchGifs';
import { ICategory } from '../interfaces';

export const GifGrid = ({ category }: ICategory) => {
  const { gifs, isLoading } = useFetchGifs(category);
  console.log({ gifs });
  return (
    <>
      <h3>{category}</h3>
      {isLoading && <h2>Cargando...</h2>}
      <div className="card-grid">
        {gifs.map((gif) => (
          <GifItem key={gif.id} {...gif} />
        ))}
      </div>
    </>
  );
};
