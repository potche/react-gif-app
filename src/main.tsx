import React from 'react';
import ReactDOM from 'react-dom/client';
import { GifApp } from './GifApp';
import './index.css';

ReactDOM.createRoot(document.getElementById('root')! as HTMLElement).render(
  <React.StrictMode>
    <GifApp />
  </React.StrictMode>,
);
