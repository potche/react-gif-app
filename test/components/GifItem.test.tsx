import React from 'react';

import { describe, expect, it, test } from 'vitest';
import { render, screen } from '@testing-library/react';
import { IGifData } from '../../src/interfaces';
import { GifItem } from '../../src/components';

describe('Pruebas en <GifItem />', () => {
  const gif: IGifData = {
    id: 'VXJWhaO7afRe',
    title: 'One Punch Man GIF',
    url: 'https://media0.giphy.com/media/VXJWhaO7afRe/giphy.gif?cid=9217b59fk7mz91mj0mxbpph0koffdt8z8lxf1akdadwpnfv2&ep=v1_gifs_search&rid=giphy.gif&ct=g',
  };

  test('debe de hacer match con el snapshot', () => {
    const { container } = render(<GifItem key={gif.id} {...gif} />);

    expect(container).toMatchSnapshot();
  });

  test('debe mostar la imagen con el URL y el ALT indicado', () => {
    const { container } = render(<GifItem key={gif.id} {...gif} />);
    // screen.debug();

    const { src, alt } = screen.getByRole<HTMLImageElement>('img');
    expect(src).toBe(gif.url);
    expect(alt).toBe(gif.title);
  });
});
