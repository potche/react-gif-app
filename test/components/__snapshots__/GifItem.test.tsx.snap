// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Pruebas en <GifItem /> > debe de hacer match con el snapshot 1`] = `
<div>
  <div
    class="card"
  >
    <img
      alt="One Punch Man GIF"
      src="https://media0.giphy.com/media/VXJWhaO7afRe/giphy.gif?cid=9217b59fk7mz91mj0mxbpph0koffdt8z8lxf1akdadwpnfv2&ep=v1_gifs_search&rid=giphy.gif&ct=g"
    />
    <p>
      One Punch Man GIF
    </p>
  </div>
</div>
`;
